bundles/org.openhab.binding.livisismarthome/src/main/resources/OH-INF/thing/channels.xml

```diff
@@ -0,0 +1,368 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="livisismarthome"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Switch Channel -->
+	<channel-type id="SwitchActuator">
+		<item-type>Switch</item-type>
+		<label>Switch</label>
+		<description>Switches the current on/off</description>
+		<category>Switch</category>
+	</channel-type>
+
+	<!-- Dimmer Channel -->
+	<channel-type id="DimmerActuator">
+		<item-type>Dimmer</item-type>
+		<label>Dimmer</label>
+		<description>Dimms the connected light</description>
+		<category>Lightbulb</category>
+	</channel-type>
+
+	<!-- RollerShutter Channel -->
+	<channel-type id="RollerShutterActuator">
+		<item-type>Rollershutter</item-type>
+		<label>Blinds Position</label>
+		<description>Controls the blinds</description>
+		<category>Blinds</category>
+		<config-description>
+			<parameter name="invert" type="boolean">
+				<label>Invert Position</label>
+				<description>When invert is true than 0 on LIVISI is UP and 100 is DOWN</description>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<!-- BooleanSwitch Channel -->
+	<channel-type id="BooleanStateActuator">
+		<item-type>Switch</item-type>
+		<label>State</label>
+		<description>Switches the state on/off</description>
+		<category>Switch</category>
+	</channel-type>
+
+	<!-- ThermostatActuator -->
+	<channel-type id="ThermostatActuator_PointTemperature">
+		<item-type>Number:Temperature</item-type>
+		<label>Target Temperature</label>
+		<description>Thermostat target temperature</description>
+		<category>Temperature</category>
+		<tags>
+			<tag>Setpoint</tag>
+			<tag>Temperature</tag>
+		</tags>
+		<state min="6" max="30" step="0.5" pattern="%.1f %unit%" readOnly="false"/>
+	</channel-type>
+	<channel-type id="ThermostatActuator_OperationMode" advanced="true">
```

Comment:
Channel type id's should also be camelCase:
```suggestion
	<channel-type id="thermostatActuatorOperationMode" advanced="true">
```

Reply:
We are talking about 80 channel-type-ids and these are not only referenced within the channels.xml...

Reply:
I knew this comment would be annoying since the PR is quite big, but the thing is that these id's cannot be changed later without breaking compatibility. So the time now will be well spent compared to the situation of having to deal with this later on. You could extract a list of all channel type id's, then prepare some scripted lines for changed them (search and replace). If you are in a Linux environment, `sed` might help. Let me know if I can help preparing something.

Reply:
Perhaps something like this (one line per channel type id when extracted):
```
grep -rl ThermostatActuator_OperationMode . | xargs sed -i 's/ThermostatActuator_OperationMode/thermostatActuatorOperationMode/g'
```
(not tested, might need some tweaking)

Reply:
and it will make all channels incompatible (for all beta testers and for all people migrating from innogysmarthome to livisismarthome). Is it really worth it (just to start the channel-type-id with a lower letter...)? Is the channel-type-id visible or configurable for a user? If not, I think it isn't worth it.

Reply:
Compatibility is already broken by renaming channel id's to conform with openHAB naming conventions. The same applies to migration from old to new binding. At least for users using managed things. For file-based configuration renaming channel type id's is not a problem, openHAB just needs to be restarted. Since this is a new binding (although replacing an old one) naming conventions should be followed. Currently it's even a mix of conventions:
- **camelCase** (rule/convention to be followed)
- **dash-separated** (seems in some bindings and makes some sense as system default channel types also use this conventions - I know, inconsistent and confusing)
- **underscore_separated**
- **PascalCase_AndUnderscores** (your naming, even internally inconsistent)

Reply:
No, only channel id's with underscores were changed. All other channel id's were already lowerCamelCase.

I will now change it, the users will love you... ;-)

Reply:
solved

Reply:
It should only affect beta testers, as there is no migration path anyway since it's technically a new binding. At least for UI configurations. File-based configurations can easily be updated. Beta testers should expect breaking changes. :) Thanks!